### PR TITLE
fix: allow 200 responses for POST requests

### PR DESCRIPTION
**Pull request by topaztee**
This PR allows for 200 response code for POST requests.

Context: https://snyk.slack.com/archives/C0141L0RUGJ/p1671033409502299?thread_ts=1670777230.330959&cid=C0141L0RUGJ

https://snyksec.atlassian.net/jira/software/c/projects/SOUL/boards/397?selectedIssue=SOUL-812

### CODE REVIEW

File: src/rulesets/rest/2022-05-25/json-api-rules/status-code-rules.ts [hunk 1]
```diff
@@ -69,13 +69,13 @@ const post2xxCodes = new ResponseRule({
...
   rule: (responseAssertions) => {
-    const validPost2xxCodes = ["201"];
+    const validPost2xxCodes = ["201", "200"];
     responseAssertions.added(
       "support the correct 2xx status codes",
       (response) => {
         if (!validPost2xxCodes.includes(response.statusCode)) {
           throw new RuleError({
-            message: `expected POST response to only support 201, not ${response.statusCode}`,
+            message: `expected POST response to only support 201 or 200, not ${response.statusCode}`,
```
Comment (jlourenc) [inline src/rulesets/rest/2022-05-25/json-api-rules/status-code-rules.ts:78]: `nit`: could we dynamically generate the string from `validPost2xxCodes`? Also, probably worth ordering them from low to high.

File: src/rulesets/rest/2022-05-25/json-api-rules/resource-object-rules.ts [hunk 2]
```diff
@@ -298,7 +298,7 @@ const locationHeader = new ResponseRule({
   name: "location header",
   matches: (responseBody, rulesContext) =>
     rulesContext.operation.method === "post" &&
-    responseBody.statusCode === "201",
+    ["200", "201"].includes(responseBody.statusCode),
```
Comment (jlourenc) [inline src/rulesets/rest/2022-05-25/json-api-rules/resource-object-rules.ts:301]: Could `validPost2xxCodes` be reused to have a single source of truth?

**Review by jlourenc — APPROVED** [review]

File: src/rulesets/rest/2022-05-25/json-api-rules/status-code-rules.ts [hunk 3]
```diff
@@ -69,13 +73,12 @@ const post2xxCodes = new ResponseRule({
...
     rulesContext.operation.method === "post",
   rule: (responseAssertions) => {
-    const validPost2xxCodes = ["201"];
     responseAssertions.added(
       "support the correct 2xx status codes",
       (response) => {
         if (!validPost2xxCodes.includes(response.statusCode)) {
           throw new RuleError({
-            message: `expected POST response to only support 201, not ${response.statusCode}`,
+            message: `expected POST response to only support ${validPost2xxCodes[0]} or ${validPost2xxCodes[1]}, not ${response.statusCode}`,
```
Comment (jlourenc) [inline src/rulesets/rest/2022-05-25/json-api-rules/status-code-rules.ts:81]: Let's not do that. Can we call `toString` on `validPost2xxcCode` instead?

Comment (topaztee) [reply src/rulesets/rest/2022-05-25/json-api-rules/status-code-rules.ts]: updated to use toString

**Pull request merged**